### PR TITLE
Improve api support

### PR DESCRIPTION
**Pull request by rande**

### CODE REVIEW

**Comment by rande** [comment]
ping @plouc  ;)

File: gitlab.go [hunk 1]
```diff
@@ -87,16 +87,23 @@ func (g *Gitlab) buildAndExecRequest(method, url string, body []byte) ([]byte, e
 	if err != nil {
 		return nil, fmt.Errorf("Client.Do error: %q", err)
 	}
+
+	if resp.StatusCode >= 400 {
```
Comment (ns3777k) [inline gitlab.go]: don't like consts? :smiley: 
`if resp.StatusCode >= http.StatusBadRequest {`

File: builds.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	FinishedAt    string        `json:"finished_at"`
+	Name          string        `json:"name"`
+	Ref           string        `json:"ref"`
+	Runner        interface{}   `json:"runner"`
+	Stage         string        `json:"stage"`
+	StartedAt     string        `json:"started_at"`
+	Status        string        `json:"status"`
+	Tag           bool          `json:"tag"`
+	User          User          `json:"user"`
+}
```
Comment (ns3777k) [inline builds.go]: `runner` and `coverage` are really `interface{}` ? i only see `null` in tests. what types can they be in real world :smile: ?

**Comment by ns3777k** [comment]
would be great if you update readme as well :pray:

File: gitlab.go (hunk 1 above)
Comment (rande) [inline gitlab.go]: It was the original code ;) I will fix it

File: builds.go (hunk 2 above)
Comment (rande) [inline builds.go]: I don't know ;) the documentation is not explicit about their values ... I can remove those fields in the mean time. So the lib will be BC when those types will be defined.

Comment (rande) [inline builds.go]: fixed

File: gitlab.go (hunk 1 above)
Comment (rande) [inline gitlab.go]: fixed

File: README.md [hunk 3]
```diff
@@ -8,33 +8,40 @@ go-gitlab-client is a simple client written in golang to consume gitlab API.
...
-	* manage user keys
-
-*	
-	###Deploy Keys [gitlab api doc](http://api.gitlab.org/deploy_keys.html)
-	* list project deploy keys
-	* add/get/rm project deploy key
-
-
-
+*
```
Comment (plouc) [inline README.md]: @rande the formatting seems broken

**Comment by rande** [comment]
@plouc I think we are ready to merge

**Comment by plouc** [comment]
@rande, I think so :)

**Comment by rande** [comment]
Thanks, I can now release: https://github.com/rande/gitlab-ci-helper

**Pull request merged**